### PR TITLE
v1 parsers and v2 parse bugs

### PR DESCRIPTION
**Pull request by lamastex**
Several parsing bugs were fixed in version 2 and new parsers done for version 1 events and gkg files with minimal unit tests. Also three collaborator information is added.

### CODE REVIEW

File: src/main/scala/com/aamend/spark/gdelt/GdeltParser.scala [hunk 1]
```diff
@@ -64,7 +150,7 @@ object GdeltParser {
       adm1Code = T(()=>tokens(38)),
       adm2Code = T(()=>tokens(39)),
       geoPoint = T(()=>actor1GeoPoint),
-      featureId = T(()=>tokens(40))
+      featureId = T(()=>tokens(42))
```
Comment (aamend) [inline src/main/scala/com/aamend/spark/gdelt/GdeltParser.scala:153]: Is that expected? I've never used feature ID in any of my analysis, so may have missed that bug - if a bug

**Comment by aamend** [comment]
I see we did not extend `implicit class GdeltSpark(dfReader: DataFrameReader)` to make V1 available through a spark context. Is that intended? 

Assuming we want to support both v1 and v2 in a same function, we may have to consider dataframe instead of dataset, and provide an option parameter as per below example

```scala
val gdeltGKG2 = spark.read.option("gdelt", "gkg2").gdelt("/path")
val gdeltGKG1 = spark.read.option("gdelt", "gkg1").gdelt("/path")
```

File: src/main/scala/com/aamend/spark/gdelt/GdeltParser.scala (hunk 1 above)
Comment (lamastex) [reply src/main/scala/com/aamend/spark/gdelt/GdeltParser.scala:153]: That's a bug. As the values differ.

Comment (AlbertNilsson) [reply src/main/scala/com/aamend/spark/gdelt/GdeltParser.scala:153]: In the old version the token(40) appears twice, first as latitude and then as feature Id. According to the documentation feature id should be on the 43rd column which corresponds to token(42).

**Review by aamend — APPROVED** [review]

**Comment by lamastex** [comment]
merging all the tightening of parsers and fully gdelt 'docs' conforming.

**Pull request merged**